workshop1/04-testing-your-code.md

```diff
@@ -41,13 +41,21 @@ This set of instructions are using [jest-dom](https://github.com/testing-library
    import { PetCard } from './PetCard';
    ```
 
-5. Add an import statement to import the `Pet` data model we created in the previous step. Again make sure that the path matches the location where the model is located relative to the `Petcard.test.tsx` file.
+5. Btw, to be able to import PetCart, you will need to export it first. Add the export keyword to the PetCard class in the `PetCard.tsx` file to achieve this:
+
+    ```typescript
+    export class PetCard extends Component<Props> {
+    ```
+
+6. Add an import statement to import the `Pet` data model we created in the previous step. Again make sure that the path matches the location where the model is located relative to the `Petcard.test.tsx` file.
 
    ```typescript
    import { Pet } from './Pet';
    ```
 
-6. Create an example pet object. We'll use this to render a `PetCard` for testing.
+7. Create an example pet object. We'll use this to render a `PetCard` for testing.
+
+    **Heads-Up**: The following mock pet has a different schema than the one found in `Pet.tsx`. What do you have to do to make this schema work?
```

Comment:
Oh, does this not work? I thought it was ok for an object of type `X` to have *extra* fields. I'll check this one, thank you!

Reply:
Yes, thank you! Well caught. TypeScript applies stricter rules to object literals, on the assumption that you're more likely to have "extra" properties in your objects because you typoed a field name.

Ref: [Excess Property Checks](https://www.typescriptlang.org/docs/handbook/interfaces.html#excess-property-checks)

I think we can fix this by either a) using a type assertion instead of typing the literal:

```tsx
const petMock = {
    ...
} as Pet;
```

Or by explicitly declaring that the interface can have additional properties:

```tsx
export default interface Pet {
    id: string,
    name: string,
    description: string,
    [propName: string]: any;
};
```

I think I'd lean towards the latter, but perhaps @stirlaf or @pviotti have thoughts?

Reply:
I'd lean towards maintaining the Pet interface as cleanly typed as possible, and use the type assertion for that temporary/mock object.  

Btw some participants may be new to Typescript, or even unwilling to learn it that day and more interested in learning React than spending time solving this somewhat pedantic typing issues. So I suggest we clearly state, even just orally, that while Typescript is arguably better than JS especially for projects that benefit from strong typing (for safety, productivity, etc), for small projects like this one the overhead of typing may outweigh its advantages. For this reason, for the purpose of this workshop, they can use Javascript or use weak typing shortcuts such as `as any` or the alternatives above.

Reply:
Great, thanks for the expert opinion! :)

Addressed in f03f173.